src/main/java/cloud/fogbow/ras/core/OrderController.java

```diff
@@ -22,6 +22,8 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
+// TODO(chico) - Question! Why is the necessity of synchronized block at activeOrdersMaps.
+//  It is already Thread Safe, right ?
```

Comment:
It is thread safe for each operation. But if you're executing a sequence of operations, then you might want that they behave as a "transaction". Synchonizing on activeOrdersMap may achieve this behavior. However, it may be the case that some (perhaps all, I can't remember now) of the synchronizations added are not needed. Please, remove the unnecessary ones and remove the TODO comment.

Reply:
Ok. done!